internal/bminventory/inventory.go

```diff
@@ -4924,12 +4924,17 @@ func (b *bareMetalInventory) V2UpdateHostInstallerArgsInternal(ctx context.Conte
 		return nil, err
 	}
 
+	allowed, err := b.authzHandler.HasAccessTo(ctx, h, auth.UpdateAction)
+	if err != nil || !allowed {
```

Comment:
See my comment on 3630:
1) no need to check err in the if because allowed is always false
2) return code should be 403 or 404

Reply:
fixed